system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -1968,16 +1952,24 @@ class CLdapClient : implements ILdapClient, public CInterface
                     LDAP_UNBIND(user_ld);
                 }
                 DBGLOG("finished LdapBind for user %s, rc=%d", username, rc);
-                if(!LdapServerDown(rc) || retries > LDAPSEC_MAX_RETRIES)
+
+                if(rc == LDAP_SUCCESS)
                     break;
-                sleep(LDAPSEC_RETRY_WAIT);
-                if(retries < LDAPSEC_MAX_RETRIES)
-                    DBGLOG("Server temporarily unreachable, retrying ...");
-                // Retrying next ldap sever, might be the same server
-                m_ldapconfig->getLdapHost(hostbuf);
+                else if(rc==LDAP_SERVER_DOWN || rc==LDAP_UNAVAILABLE)
+                {
+                    m_ldapconfig->rejectHost(hostbuf);
```

Comment:
Changed in a third different way?  I see that it isn't looping through the hosts in the same way, but I would expect the code to be changed to make it more similar.

Reply:
Removed the unnecessary check for LDAP_SUCCESS, but note that the rejectHost call makes sense inside of the retry loop, since there isn't an outer loop going through numHosts

Reply:
Why isn't there an outer loop?  I think it was missed as part of the changes in HPCC-16179.

It means that only 3 servers will be checked, but that shouldn't cause problems in practice.